dart/lib/src/sentry_client.dart

```diff
@@ -38,6 +38,19 @@ const _defaultIpAddress = '{{auto}}';
 @visibleForTesting
 String get defaultIpAddress => _defaultIpAddress;
 
+@internal
+typedef SdkLifecycleCallback<T extends SdkLifecycleEvent> = FutureOr<void>
+    Function(T event);
+
+@internal
+abstract class SdkLifecycleEvent {}
+
+@internal
+class OnBeforeCaptureLog extends SdkLifecycleEvent {
+  final SentryLog log;
+  OnBeforeCaptureLog(this.log);
+}
```

Comment:
Unsure whether we should move this to its own file, as we add more and more lifecycle events this is gonna increase and maybe overload the sentry_client file